plugins/fastem_roa_import_export.py

```diff
@@ -0,0 +1,341 @@
+                        else:
+                            roas[fieldname.value] = row[fieldname.value]
+                    if project_name in data:
+                        data[project_name].append(roas)
+                    else:
+                        data[project_name] = [roas]
+                # Secondly create project control and update the roa's and roc's
+                # data accordingly
+                for project_name, roas in data.items():
+                    project_ctrl = self._project_list_controller._add_project(
+                        None, name=project_name
+                    )
+                    for roa in roas:
+                        roa_name = roa[CSVFieldnames.ROA_NAME.value]
+                        roa_coordinates = (
+                            float(roa[CSVFieldnames.ROA_COORDINATES_L.value]),
+                            float(roa[CSVFieldnames.ROA_COORDINATES_T.value]),
+                            float(roa[CSVFieldnames.ROA_COORDINATES_R.value]),
+                            float(roa[CSVFieldnames.ROA_COORDINATES_B.value]),
+                        )
+                        roc_2_name = roa[CSVFieldnames.ROC_2_NAME.value]
+                        roc_2_coordinates = (
+                            float(roa[CSVFieldnames.ROC_2_COORDINATES_L.value]),
+                            float(roa[CSVFieldnames.ROC_2_COORDINATES_T.value]),
+                            float(roa[CSVFieldnames.ROC_2_COORDINATES_R.value]),
+                            float(roa[CSVFieldnames.ROC_2_COORDINATES_B.value]),
+                        )
+                        roc_3_name = roa[CSVFieldnames.ROC_3_NAME.value]
+                        roc_3_coordinates = (
+                            float(roa[CSVFieldnames.ROC_3_COORDINATES_L.value]),
+                            float(roa[CSVFieldnames.ROC_3_COORDINATES_T.value]),
+                            float(roa[CSVFieldnames.ROC_3_COORDINATES_R.value]),
+                            float(roa[CSVFieldnames.ROC_3_COORDINATES_B.value]),
+                        )
+                        project_ctrl.regions_calib_2.value[int(roc_2_name)].name.value = roc_2_name
+                        project_ctrl.regions_calib_2.value[
+                            int(roc_2_name)
+                        ].coordinates.value = roc_2_coordinates
+                        project_ctrl.regions_calib_3.value[int(roc_3_name)].name.value = roc_3_name
+                        project_ctrl.regions_calib_3.value[
+                            int(roc_3_name)
+                        ].coordinates.value = roc_3_coordinates
+                        roa_ctrl = project_ctrl._on_btn_roa(None, name=roa_name)
+                        roa_ctrl.model.coordinates.value = roa_coordinates
+        except Exception as ex:
+            logging.exception(ex)
+            wx.MessageBox(
+                f"Importing {pathname} failed, please see log.", "Error", wx.OK | wx.ICON_ERROR
```

Comment:
You *could* pass "ex" here, so that the failure is less mysterious. Maybe it means nothing to the user... or maybe it already give them a clue. I don't think it can be too much information.